accepted/0000-basic-orchestration.md

```diff
@@ -0,0 +1,125 @@
+- Feature Name: initial_salt_orchestration_support
+- Start Date: 2020-07-03
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Implement basic support for Salt Orchestration in Uyuni.
+
+# Motivation
+[motivation]: #motivation
+
+The initial use case is improving clustering support. Use cases requiring orchestration:
+- Use temporary SSH keys on CaaSP cluster nodes: a SSH key must be generated, authorized on all the nodes on the cluster, then `skuba` is invoked to execute a cluster operation requiring the use of the temporary SSH key, then finally the key is removed from the nodes.
+- Setup a new CaaSP cluster from on a group of machines managed by Uyuni. Some of the steps involved are: setting up a load balancer (optional), authorizing a SSH key on the target machines, changing the `sudoers` file, loading the SSH key in the `ssh-agent`, running `skuba init` and `skuba node bootstrap` on the management node, add any nodes to the cluster by running `skuba node join`.
+- Protected sensitive cluster packages using locks. E.g. if locks are used to protected the CaaSP packages from accidental upgrade or remove ( from Uyuni, Salt or command line), the locks have to be removed before doing upgrade and then reinstated after the upgrade operation.
+
+In the initial version, the scope of the orchestration support will be limited to clustering. In later iterations it can be expanded to support other uses cases.
+
+# Detailed design
+[design]: #detailed-design
+
+The `salt-netapi` library already support calling Salt runner. 
+
+The Uyuni actions must be enhanced to make Salt runner calls besides minions calls.
+
+## Action enhancements.
+
+A runner action won't need a collection of `ServerAction` objects because the action is not executed on a set of minions directly.
+
+Salt supports asynchronous runner calls but it doesn't allow attaching metadata like `action-id`. Therefore the `jid` returned by Salt must be stored in the database in order to match the return event to the originating action. 
```

Comment:
What would be pros and cons of extending Salt so that runner calls can also have arbitrary metadata?